hoomd/md/methods.py

```diff
@@ -225,251 +225,79 @@ class npt(_Method):
+        integrator = integrate.NPT(filter=all, tau=1.0, kT=0.65, tauS = 1.2, S=2.0)
         # orthorhombic symmetry
-        integrator = integrate.npt(group=all, tau=1.0, kT=0.65, tauP = 1.2, P=2.0, couple="none")
+        integrator = integrate.NPT(filter=all, tau=1.0, kT=0.65, tauS = 1.2, S=2.0, couple="none")
         # tetragonal symmetry
-        integrator = integrate.npt(group=all, tau=1.0, kT=0.65, tauP = 1.2, P=2.0, couple="xy")
+        integrator = integrate.NPT(filter=all, tau=1.0, kT=0.65, tauS = 1.2, S=2.0, couple="xy")
         # triclinic symmetry
-        integrator = integrate.npt(group=all, tau=1.0, kT=0.65, tauP = 1.2, P=2.0, couple="none", rescale_all=True)
+        integrator = integrate.NPT(filter=all, tau=1.0, kT=0.65, tauS = 1.2, S=2.0, couple="none", rescale_all=True)
     """
-    def __init__(self, group, kT=None, tau=None, S=None, P=None, tauP=None, couple="xyz", x=True, y=True, z=True, xy=False, xz=False, yz=False, all=False, nph=False, rescale_all=None, gamma=None):
-
-        # check the input
-        if (kT is None or tau is None):
-            if nph is False:
-                hoomd.context.current.device.cpp_msg.error("integrate.npt: Need temperature T and thermostat time scale tau.\n")
-                raise RuntimeError("Error setting up NPT integration.")
-            else:
-                # use dummy values
-                kT=1.0
-                tau=1.0
-
-        if (tauP is None):
-                hoomd.context.current.device.cpp_msg.error("integrate.npt: Need barostat time scale tauP.\n")
-                raise RuntimeError("Error setting up NPT integration.")
-
-        # initialize base class
-        _Method.__init__(self)
-
-        # setup the variant inputs
-        kT = hoomd.variant._setup_variant_input(kT)
-
-        # If P is set
-        if P is not None:
-            self.S = [P,P,P,0,0,0]
-        else:
-            # S is a stress, should be [xx, yy, zz, yz, xz, xy]
-            if S is not None and len(S)==6:
-                self.S = S
-            else:
-                raise RuntimeError("Unrecognized stress tensor form")
-
-        S = [hoomd.variant._setup_variant_input(self.S[i]) for i in range(6)]
+    def __init__(self, filter, kT=None, tau=None, S=None, tauS=None, couple="xyz", box_dof=[True,True,True,False,False,False], rescale_all=False, gamma=0.0):
 
-        Svar = [S[i].cpp_variant for i in range(6)]
 
-        # create the compute thermo for half time steps
-        if group is hoomd.context.current.group_all:
-            group_copy = copy.copy(group)
-            group_copy.name = "__npt_all"
-            thermo_group = hoomd.compute.thermo(group_copy)
-            thermo_group.cpp_compute.setLoggingEnabled(False)
-        else:
-            thermo_group = hoomd.compute._get_unique_thermo(group=group)
+        # store metadata
+        param_dict = ParameterDict(
+            filter=_ParticleFilter,
+            kT=Variant,
```

Comment:
`kT` is the right name for the parameter, but C++ still exports the member as `T`. Have you tested if you can set `kT` after attaching? I think that test will fail unless you export the `setT/getT` as `kT` from C++.